### PR TITLE
updates for container registry secrets/config

### PR DESCRIPTION
**Pull request by dustymabe**
```
commit e44a8c2c84fe7a9373064e7f226c4712dcb0f3ee
Author: Dusty Mabe <dusty@dustymabe.com>
Date:   Tue Sep 27 13:11:43 2022 -0400

    jobs/release: drain registry push information into the config
    
    The information about where to push the containers shouldn't be
    hardcoded in the job. Let's drain that into the config and act
    accordingly.

commit 910bd47c5923e02c95e513035a1a1f2dda6e202a
Author: Dusty Mabe <dusty@dustymabe.com>
Date:   Tue Sep 27 13:05:53 2022 -0400

    rename fedora-push-registry-secret to oscontainer-push-registry-secret
    
    This secret is used to push the oscontainer to the registry. For FCOS
    it's the fedora registry, but for RHCOS it will be a different registry.
    Let's make the naming more generic.

commit e6a4519d101f38f3551ee76d66f3461ed559df12
Author: Dusty Mabe <dusty@dustymabe.com>
Date:   Tue Sep 27 09:42:50 2022 -0400

    convert container registry secrets to use kubernetes-credentials plugin
    
    This allows the credentials to get magically populated in Jenkins
    without having to have mounts and casc config for them. This is using
    the https://github.com/jenkinsci/kubernetes-credentials-provider-plugin

commit 9dfdc1784afc54cdd5f9407aeaaa56c0a8da2a44
Author: Dusty Mabe <dusty@dustymabe.com>
Date:   Tue Sep 27 13:52:34 2022 -0400

    HACKING: update docs for additional-root-ca-cert secret
    
    This updates the instructions to be a little less manual by not
    requiring the user to base64 encode the file themselves and also
    gives them the exact commands to run.
```

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -273,17 +273,38 @@ follows:
 
 ### [PROD] Create quay.io Fedora image push secret
 
-This secret is used to push the oscontainer and others to Quay.io
+This secret is used to push the oscontainer and others to Quay.io.
+The secret can be obtained from the `fedora-push-registry-secret` in BitWarden.
 
-1. Obtain the file `fedora-push-registry-secret` from BitWarden.
-2. Run: `$ oc create secret generic fedora-push-registry-secret --from-file=dockercfg=fedora-push-registry-secret`.
+After obtaining the secret data you can create the kubernetes secret via:
```
Comment (jlebon) [inline HACKING.md:279]: Minor: Kubernetes

(Here and in other new hunks.)

File: HACKING.md [hunk 2]
```diff
@@ -273,17 +273,38 @@ follows:
...
+
+```
+oc create secret generic fedora-push-registry-secret \
+    --from-literal=filename=dockercfg \
+    --from-file=data=fedora-push-registry-secret
+oc label secret/fedora-push-registry-secret \
+    jenkins.io/credentials-type=secretFile
+oc annotate secret/fedora-push-registry-secret \
+    jenkins.io/credentials-description="Registry push secret for pushing the CoreOS OSContainer"
```
Comment (jlebon) [inline HACKING.md:288]: Minor: feels like "Registry push secret for CoreOS OSContainer" would be sufficient? (And similarly below.)

File: manifests/jenkins.yaml [hunk 3]
```diff
@@ -159,15 +159,6 @@ objects:
           - name: coreos-x86-64-builder
             mountPath: /var/run/secrets/coreos-x86-64-builder
             readOnly: true
-          - name: oscontainer-secret
-            mountPath: /var/run/secrets/oscontainer-secret
-            readOnly: true
```
Comment (jlebon) [inline manifests/jenkins.yaml:164]: I'm confused by this commit. It moves over the cosa and Fedora secrets but here we also delete the secret for the legacy location.

Hmm, are we missing a section for it in the HACKING.md to set that up?

File: config.yaml [hunk 4]
```diff
@@ -30,6 +30,10 @@ source_config:
 # remove to disable S3 uploads
 s3_bucket: fcos-builds
 
+registry_repos:
+  oscontainer: quay.io/fedora/fedora-coreos
+  old_oscontainer: quay.io/coreos-assembler/fcos
```
Comment (jlebon) [inline config.yaml:35]: Given that the old location is a temporary thing, it seems odd to make this a full-blown object. Also, "old oscontainer" might be confused to mean the old oscontainer format.

WDYT about simply:

```yaml
oscontainer_repo: quay.io/fedora/fedora-coreos
oscontainer_old_repo: quay.io/coreos-assembler/fcos
```

And then we can eventually drop support for `oscontainer_old_repo`.

File: manifests/jenkins.yaml (hunk 3 above)
Comment (dustymabe) [reply manifests/jenkins.yaml:164]: ehh yeah. That secret is going away in like a month so I elected to leave it out of `HACKING.md`. It already wasn't in there.

Comment (jlebon) [reply manifests/jenkins.yaml:164]: Ack gotcha. I think ideally we'd have it there for completeness but OK as is.

File: config.yaml (hunk 4 above)
Comment (jlebon) [reply config.yaml:35]: OK that makes sense. Re. name, maybe just `oscontainer_old`?

Comment (dustymabe) [reply config.yaml]: Done

**Review by jlebon — APPROVED** [review]
LGTM!

**Pull request merged**